### PR TITLE
Allow embedder clients to provide a resource context

### PR DESCRIPTION
**Pull request by abhishekamit**
Closes flutter/flutter#14263

### CODE REVIEW

File: shell/platform/embedder/embedder.cc [hunk 1]
```diff
@@ -125,6 +125,14 @@ FlutterResult FlutterEngineRun(size_t version,
     };
   }
 
+  std::function<bool()> make_resource_current_callback = nullptr;
+  if (config->open_gl.make_resource_current != nullptr) {
```
Comment (chinmaygarde) [inline shell/platform/embedder/embedder.cc]: This needs to be wrapped with the SAFE_ACCESS macro.

File: shell/platform/embedder/embedder.cc [hunk 2]
```diff
@@ -125,6 +125,14 @@ FlutterResult FlutterEngineRun(size_t version,
...
+    make_resource_current_callback =
+        [ ptr = config->open_gl.make_resource_current, user_data ]()->intptr_t {
```
Comment (chinmaygarde) [inline shell/platform/embedder/embedder.cc]: The `-> intptr_t` will introduce an unnecessary cast. You could just skip it.

File: shell/platform/embedder/embedder.cc (hunk 1 above)
Comment (abhishekamit) [reply shell/platform/embedder/embedder.cc]: Done

File: shell/platform/embedder/embedder.cc (hunk 2 above)
Comment (abhishekamit) [reply shell/platform/embedder/embedder.cc]: Done

**Pull request merged**